src/diagnostics/MOM_wave_speed.F90

```diff
@@ -945,19 +1084,32 @@ subroutine wave_speeds(h, tv, G, GV, US, nmodes, cn, CS, w_struct, u_struct, u_s
           endif
         endif
 
+        if (nonBous) then
+          ! Note that dSpVxh_sum is negative for stable stratification.
+          cg1_est = H_to_pres * abs(dSpVxh_sum)
+        else
+          cg1_est = g_Rho0 * drxh_sum
+        endif
+
         !   Find gprime across each internal interface, taking care of convective
         ! instabilities by merging layers.
-        if (g_Rho0 * drxh_sum > cg1_min2) then
+        if (cg1_est > cg1_min2) then
           ! Merge layers to eliminate convective instabilities or exceedingly
           ! small reduced gravities.  Merging layers reduces the estimated wave speed by
           ! (rho(2)-rho(1))*h(1)*h(2) / H_tot.
-          if (use_EOS) then
+          if (use_EOS .and. (.not.nonBous)) then
```

Comment:
`.and. (.not.nonBous)` causes the code to switch to layer mode

Reply:
Well spotted.  This is easily corrected.